corehq/apps/locations/models.py

```diff
@@ -436,19 +436,27 @@ def delete(self, *args, **kwargs):
     full_delete = delete
 
     def get_descendants(self, include_self=False, **kwargs):
+        include_archived = kwargs.pop('include_archived', True)
         if include_self:
             where = Q(domain=self.domain, id=self.id)
         else:
             where = Q(domain=self.domain, parent_id=self.id)
-        return SQLLocation.objects.get_descendants(
-            where, **kwargs
-        )
+        if include_archived:
+            locations = SQLLocation.objects
+        else:
+            locations = SQLLocation.active_objects
+
+        return locations.get_descendants(where, **kwargs)
 
     def get_ancestors(self, include_self=False, **kwargs):
+        include_archived = kwargs.pop('include_archived', True)
```

Comment:
I'm actually not sure it's useful to be able to specify in this context.  If a location is active, none of its ancestors are archived anyways.  If you're calling `get_ancestors` on an archived location, you'll want the archived ancestors too (you're probably doing something like [unarchiving](https://github.com/dimagi/commcare-hq/blob/330c8e6a1bde78914342bb69fa29c17751f1aaa8/corehq/apps/locations/models.py#L578-L578) that location)